doc/decisions/map_lazyloading.md

```diff
@@ -0,0 +1,58 @@
+# Map Lazyloading
+
+## Problem
+
+The application's main component is the map, which consists of multiple layers and even more objects.
+One more thing to consider is that the map consists not only of the data for the current date, but also for the past and future dates e.g. harvest time, planting history.
+However, not all layers are visible simultaneously, nor are they required during the initial loading of the application.
+
+Loading all the data at once during the initial load may result in slower performance, therefore a strategy must be developed to lazyload data on the map to optimize its performance.
+In other words, the data which is not required for the user to start working with the map should not be loaded during the startup.
+
+## Constraints
+
+N/A
```

Comment:
We want to avoid on-screen loading, as this is very bad for offline usage. So your goal is to search how to cluster the data in layers, so that everything works smoothly as long as not too many layers are activated. Of course our goal is to implement the architecture exactly as you propose in the feasibility study. Thus it is important that we **first** clarify objectives etc. of what exactly you study the feasibility of.